src/test/scala/org/constellation/trust/WalkTest.scala

```diff
@@ -0,0 +1,24 @@
+package org.constellation.trust
+
+import cats.effect.{ConcurrentEffect, ContextShift, IO}
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.ExecutionContext
+
+class WalkTest extends AnyFreeSpec with Matchers {
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+  implicit val effect: ConcurrentEffect[IO] = IO.ioConcurrentEffect(contextShift)
+  val generator = new DataGenerator[IO]()
+  "Should not create a stack overflow" in {
+
+    /*
+    Doesn't seem to re-create issue even with 10k+ nodes
```

Comment:
If we can't see it on a cluster than it was probably a bug somewhere else. I would move forward incorporating into the L0 or snapshot creation flow in tessellation if this is the case